src/scripts/main.js

```diff
@@ -1 +1,70 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeOut = setTimeout(() => {
+    reject(new Error('First promise was rejected in 3 seconds if not clicked'));
+  }, 3000);
+
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      clearTimeout(timeOut);
+      resolve('First promise was resolved on a left click in the document');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+  });
+});
```

Comment:
For the second promise, you are adding two event listeners for 'click' and 'contextmenu'. However, both listeners resolve the promise if they are triggered. It means that the promise will be resolved on the first event that occurs. If you want the promise to be resolved only when both events have occurred, you need to adjust your logic accordingly.